testsuite/MDAnalysisTests/analysis/test_hydrogenbondautocorrel_deprecated.py

```diff
@@ -0,0 +1,331 @@
+                 acceptors=oxygens,
+                 donors=nitrogens,
+                 bond_type='intermittent',
+                 exclusions=excl_list2,
+                 sample_time=0.06,
+        )
+
+    def test_bond_type_VE(self, u, hydrogens, oxygens, nitrogens):
+        with pytest.raises(ValueError):
+            HBAC(u,
+                 hydrogens=hydrogens,
+                 acceptors=oxygens,
+                 donors=nitrogens,
+                 bond_type='marzipan',
+                 exclusions=(np.arange(len(hydrogens)), np.array(range(
+                     len(oxygens)))),
+                 sample_time=0.06,
+        )
+
+    def test_solve_before_run_VE(self, u, hydrogens, oxygens, nitrogens):
+        hbond = HBAC(u,
+                     hydrogens=hydrogens,
+                     acceptors=oxygens,
+                     donors=nitrogens,
+                     bond_type='continuous',
+                     sample_time=0.06,
+        )
+        with pytest.raises(ValueError):
+            hbond.solve()
+
+    @mock.patch('MDAnalysis.coordinates.TRZ.TRZReader._read_frame')
+    def test_unslicable_traj_VE(self, mock_read, u, hydrogens, oxygens, nitrogens):
+        mock_read.side_effect = TypeError
+
+        with pytest.raises(ValueError):
+            HBAC(
+                u,
+                hydrogens=hydrogens,
+                acceptors=oxygens,
+                donors=nitrogens,
+                bond_type='continuous',
+                sample_time=0.06
+        )
+
+    def test_repr(self, u, hydrogens, oxygens, nitrogens):
+        hbond = HBAC(u,
+                     hydrogens=hydrogens,
+                     acceptors=oxygens,
+                     donors=nitrogens,
+                     bond_type='continuous',
+                     sample_time=0.06,
+        )
+        assert isinstance(repr(hbond), str)
+
+    def test_deprecation_warning(self, u, hydrogens, oxygens, nitrogens):
+        wmsg = ('`HydrogenBondAutoCorrel` is deprecated!\n'
+                '`HydrogenBondAutoCorrel` will be removed in release 3.0.0.\n'
+                'The class was moved to MDAnalysis.analysis.hbonds.hbond_autocorrel.')
+        with pytest.warns(DeprecationWarning, match=wmsg):
+            HBAC(
+                u,
+                hydrogens=hydrogens,
+                acceptors=oxygens,
+                donors=nitrogens,
+                bond_type='continuous',
+                sample_time=0.06
+            )
+
+
+
+def test_find_donors(u_water):
+    H = u_water.select_atoms('name H*')
+
+    D = hbonds.find_hydrogen_donors(H)
+
+    assert len(H) == len(D)
+    # check each O is bonded to the corresponding H
+    for h_atom, o_atom in zip(H, D):
+        assert o_atom in h_atom.bonded_atoms
+
+
+def test_donors_nobonds():
+    u = mda.Universe(XYZ_mini)
+
+    with pytest.raises(mda.NoDataError):
+        hbonds.find_hydrogen_donors(u.atoms)
+
+
+def test_find_hydrogen_donors_deprecation_warning(u_water):
```

Comment:
@orbeckst is there any point in having most of the hbond_autocorrel tests duplicated here? They are pretty expensive iirc, if you're happy with it, we can just move these two warning tests over to the other hbond_autocorrel tests?